kube-derive/src/custom_resource.rs

```diff
@@ -374,6 +374,7 @@ pub(crate) fn derive(input: proc_macro2::TokenStream) -> proc_macro2::TokenStrea
                 .with(|s| {
                     s.inline_subschemas = true;
                     s.meta_schema = None;
+                    s.option_nullable = false;
```

Comment:
oh, this is all that is needed? that is pretty nice.

so iiuc, we are relying on `required` bools to do the bulk of the heavy lifting?

Reply:
I touched on it a bit in the issue. I think this change should be fine even though it touches basically every CRD generated.

`nullable: true` from my understanding just means that k8s keeps `foo: null` fields around instead of removing them. With additional interaction with the `default` field (which I am not 100% sure how that interacts in all combinations of nullable and default). Nullable also does not mean that the field is required. Serde also treats a `null` and a `missing` value the same when deserializing a non-required Option.

So overall I think this should be fine and not break because there is (from what I saw) no way to set the `default` field on the schema (except through a custom `JsonSchema` impl, which wouldnt change) and serde should still be able to serialize/deserialize just fine, assuming no `Option` is marked as required.